From jirislaby@gmail.com  Wed Mar 11 14:01:52 2009
From: Jiri Slaby <jirislaby@gmail.com>
Date: Wed, 11 Mar 2009 21:47:39 +0100
Subject: USB: misc/vstusb, fix lock imbalance
To: gregkh@suse.de
Cc: Jiri Slaby <jirislaby@gmail.com>
Message-ID: <1236804460-1432-4-git-send-email-jirislaby@gmail.com>


Make sure we don't leak locked vstdev->lock in vstusb_write. Unlock
properly on one fail path.

Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/vstusb.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/misc/vstusb.c
+++ b/drivers/usb/misc/vstusb.c
@@ -401,6 +401,7 @@ static ssize_t vstusb_write(struct file 
 	}
 
 	if (copy_from_user(buf, buffer, count)) {
+		mutex_unlock(&vstdev->lock);
 		dev_err(&dev->dev, "%s: can't copy_from_user\n", __func__);
 		retval = -EFAULT;
 		goto exit;
